internal/gatewayapi/listener.go

```diff
@@ -96,7 +96,15 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 
 			// Add the listener to the Xds IR
 			servicePort := &protocolPort{protocol: listener.Protocol, port: int32(listener.Port)}
-			containerPort := servicePortToContainerPort(servicePort.port)
+
+			containerPort := servicePort.port
+			if !(resources.EnvoyProxy != nil && resources.EnvoyProxy.Spec.Provider != nil &&
```

Comment:
just out of curiosity, is there any better way to improve this looong hideous nested nil check?

Reply:
refer to https://github.com/envoyproxy/gateway/blob/main/api/v1alpha1/envoygateway_helpers.go#L80

make it as a common helper to use